packages/@aws-cdk/aws-lambda-event-sources/lib/stream.ts

```diff
@@ -21,6 +21,47 @@ export interface StreamEventSourceProps {
    */
   readonly batchSize?: number;
 
+  /**
+   * If the function returns an error, split the batch in two and retry.
+   *
+   * @default false
+   */
+  readonly bisectBatchOnFunctionError?: boolean;
+
+  /**
+   * An Amazon SQS queue or Amazon SNS topic destination for discarded records.
+   *
+   * @default discarded records are ignored
+   */
+  readonly destinationConfig?: lambda.CfnEventSourceMapping.DestinationConfigProperty | IResolvable | undefined;
+
+  /**
+   * The maximum age of a record that Lambda sends to a function for processing.
+   * Valid Range:
+   * * Minimum value of 60
+   * * Maximum value of 604800
+   *
+   * @default 604800
+   */
+  readonly maximumRecordAgeInSeconds?: number;
```

Comment:
Name this `maxRecordAge` and switch the type to `Duration`

Reply:
These changes would ripple into the lambda.generated.ts.  I think those come out of the cfnspec's.  Would you suggest leaving that stable and converting the duration into seconds on our side?  That would indeed allow me to clean up a bunch of linter exceptions that I had to put in

Reply:
This is cleaned up now